zaga-web/src/app/components/creatework-order/creatework-order.component.ts

```diff
@@ -192,113 +257,223 @@ export class CreateworkOrderComponent implements OnInit {
-        });
+      if (material.material.id == this.idOfEditingMaterial) {
+        material.material.id = this.selectedMaterial.split("&")[0];
+        material.material.name = this.selectedMaterial.split("&")[1];
+        material.quantity = this.quantityEntered;
+        material.material.unit = this.selectedUnit;
+        this.getArea();
+        material.quantityPerHectar = this.quantityEntered / this.crop.area;
+        console.log(this.woMaterials);
       }
     });
-    this.material = new Material();
+    this.material = new SpentMaterial();
+    this.selectedMaterial = null;
+    this.selectedUnit = null;
+    this.quantityEntered = null;
     this.editingMaterial = false;
+    this.closebuttonMaterialModal.nativeElement.click();
   }
 
-  addWorkOrder() {
-    this.workOrder.start = "2020-09-16";
-    this.workOrder.end = "2020-09-17";
-    this.workOrder.cropId = "152afb01-a708-4659-9805-bd83f8f742bb";
-    this.workOrder.machines = this.woMachines;
-    this.workOrder.workers = this.employees;
-    this.workOrder.materials = this.woMaterials;
-    this.workOrder.responsibleId = this.nameFC.value.userId;
-
-    this.workOrderService.addWorkOrder(this.workOrder).subscribe(
-      (data) => {
-        this.toastr.success("Uspešno kreiran radni nalog.");
-      },
-      (error) => {
-        this.toastr.error("Radni nalog nije kreiran.");
+  addWorkOrder(valid) {
+    this.clickAddWorkOrder = true;
+    if (valid) {
+      var dateStartToAdd = "";
+      var dateEndToAdd = "";
+      if (this.workOrder.start != undefined) {
+        if (this.workOrder.start.month < 10) {
+          this.workOrder.start.month = "0" + this.workOrder.start.month;
+        }
+        if (this.workOrder.start.day < 10) {
+          this.workOrder.start.day = "0" + this.workOrder.start.day;
+        }
+        dateStartToAdd =
+          this.workOrder.start.year +
+          "-" +
+          this.workOrder.start.month +
+          "-" +
+          this.workOrder.start.day;
+      }
+      if (this.workOrder.end != undefined) {
+        if (this.workOrder.end.month < 10) {
+          this.workOrder.end.month = "0" + this.workOrder.end.month;
+        }
+        if (this.workOrder.end.day < 10) {
+          this.workOrder.end.day = "0" + this.workOrder.end.day;
+        }
+        dateEndToAdd =
+          this.workOrder.end.year +
+          "-" +
+          this.workOrder.end.month +
+          "-" +
+          this.workOrder.end.day;
       }
-    );
 
-    console.log(this.workOrder);
+      this.workOrder.start = dateStartToAdd;
+      this.workOrder.end = dateEndToAdd;
+
+      this.workOrder.machines = this.woMachines;
+      this.workOrder.assignedUsers = this.employees;
+      this.workOrder.materials = this.woMaterials;
+
+      this.workOrder.responsibleId = this.nameFC.value.userId;
+
+      this.workOrderService.addWorkOrder(this.workOrder).subscribe(
+        (data) => {
+          this.toastr.success("Uspešno kreiran radni nalog.");
```

Comment:
Steps were taken:
Entered header information, except dates and responsible person - no workers, machines, materials
Got validation messages for dates and responsible person
Entered dates but not responsible person
Application fast set again validation messages for dates, routed me to /workOrder, sent me a confirmation message about a created work order, but the list was empty

Expected steps:
Responsible person is mandatory, should notify user about that validation